### PR TITLE
Stronger typing of upload code.

### PR DESCRIPTION
**Pull request by jmchilton**
## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. tox -e mypy

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/actions/library.py [hunk 1]
```diff
@@ -72,7 +74,7 @@ class LibraryActions:
     Mixin for controllers that provide library functionality.
     """
 
-    def _upload_dataset(self, trans, library_id, folder_id, replace_dataset=None, **kwd):
+    def _upload_dataset(self, trans, library_id: str, folder_id: str, replace_dataset: Optional[LibraryDataset] = None, **kwd):
```
Comment (davelopez) [inline lib/galaxy/actions/library.py:77]: Looks like the `library_id` parameter is not really used in `_upload_dataset` and further down in `_create_folder` maybe we can safely delete them?

**Review by davelopez — APPROVED** [review]
:+1:

File: lib/galaxy/actions/library.py (hunk 1 above)
Comment (jmchilton) [reply lib/galaxy/actions/library.py:77]: Everything is green so I'm going to merge this - but I will follow up on this point! Thanks for the catch.

Comment (jmchilton) [reply lib/galaxy/actions/library.py:77]: It is in my deferred data branch (https://github.com/galaxyproject/galaxy/pull/12533/commits/f14c978e7e99f8512ba5b4e09ffc0ed551b21a09) - once things are green there again I'll PR it or PR it as part of other changes.

**Pull request merged**